airflow/providers/hashicorp/_internal_client/vault_client.py

```diff
@@ -318,15 +321,36 @@ def _auth_azure(self, _client: hvac.Client) -> None:
             )
 
     def _auth_aws_iam(self, _client: hvac.Client) -> None:
-        if self.auth_mount_point:
-            _client.auth.aws.iam_login(
-                access_key=self.key_id,
-                secret_key=self.secret_id,
-                role=self.role_id,
-                mount_point=self.auth_mount_point,
-            )
+        if self.key_id and self.secret_id:
+            auth_args = {
+                "access_key": self.key_id,
+                "secret_key": self.secret_id,
+                "role": self.role_id,
+            }
         else:
-            _client.auth.aws.iam_login(access_key=self.key_id, secret_key=self.secret_id, role=self.role_id)
+            import boto3
+
+            if self.role_arn:
+                sts_client = boto3.client("sts")
+                credentials = sts_client.assume_role(RoleArn=self.role_arn, RoleSessionName="airflow")
```

Comment:
@pankajastro - we should consider exposing [assume_role_kwargs](https://github.com/apache/airflow/blob/providers-amazon/8.17.0/airflow/providers/amazon/aws/hooks/base_aws.py#L278) as a `backend_kwarg`.

This is currently available as an [AWS Connection Extra parameter](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/connections/aws.html#default-connection-ids).

Reply:
Could you please share an example of AIRFLOW__SECRETS__BACKEND_KWARGS?

Reply:
Please see the [Request Syntax](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sts/client/assume_role.html) section for example kwargs.

I imagine it would be a dictionary that you would pass to `AIRFLOW__SECRETS__BACKEND_KWARGS`, like one does today in the AWS connection.

Reply:
wdyt https://github.com/apache/airflow/pull/39279

```
AIRFLOW__SECRETS__BACKEND_KWARGS='{"kv_engine_version": 1, "mount_point": "kv", "variables_path": "airflow", "url": "http://127.0.0.0:8200/", "auth_type": "aws_iam", "assume_role_kwargs": {"RoleArn": "arn:aws:iam::1234567890000:role/hashicorp-aws-iam", "RoleSessionName": "airflow", "DurationSeconds": 900}}'
```

Reply:
Yes that looks good (assuming that it works). You wouldn't need a `role_arn` arg in this case.

Reply:
It's included in the `assume_role_kwargs` parameter. I've shared an example for reference ^^

Reply:
Agreed. I am saying you no longer need a dedicated `role_arn` backend kwarg. It would be a part of `assume_role_kwargs`.